lhotse/dataset/collation.py

```diff
@@ -404,7 +410,14 @@ def read_audio_from_cuts(
     cuts: Iterable[Cut], executor: Optional[Executor] = None
 ) -> List[torch.Tensor]:
     map_fn = map if executor is None else executor.map
-    return list(map_fn(_read_audio, cuts))
+    audios = []
+    for idx, audio_or_none in enumerate(map_fn(_read_audio, cuts)):
+        if audio_or_none is None:
+            del cuts[cuts[idx].id]  # delete the problematic cut
```

Comment:
I think this code is technically not correct, as `cuts` may be an `Iterable` (i.e. it may be not indexable). Instead of trying to delete the cuts, I would recommend changing the return type to ``List[Optional[torch.Tensor]]`` and have the caller of this function check each tensor for `None` to see if it was read successfully.

Reply:
Actually, that won't solve the issue, disregard that remark. It's better to change the `cuts` type annotation to `CutSet`. BTW does `CutSet` support using `del`? I never checked..

Reply:
>  BTW does `CutSet` support using `del`? I never checked..

No, I thought it is a dict.  It doesn't support `del`.

I saw in the docs, said "Operations on cut sets are not mutating...", so should I add a delete operator to it ?

Reply:
I don't like that option, there has to be a cleaner way to do it. Give me some time to think about it.